pkg/flags/flag.go

```diff
@@ -162,3 +176,23 @@ func IsSet(fs *flag.FlagSet, name string) bool {
 	})
 	return set
 }
+
+// Op represents an Operation that flags can execute.
+type Op struct {
+	noConflict bool
+}
+
+// OpOption configures flags operations.
+type OpOption func(*Op)
+
+// WithNoConflict configures no-conflict mode.
+// Expects error on conflicting flags.
+func WithNoConflict() OpOption {
```

Comment:
is it worth making this optional? I feel like it should always error out on shadowing